tests/unit/test_charm.py

```diff
@@ -427,14 +367,8 @@ def test_given_content_of_config_file_not_changed_when_pebble_ready_then_config_
         )
         self._create_database_relation_and_populate_data()
         self.harness.container_pebble_ready("amf")
-        patch_push.assert_called_once_with(
-            path="/support/TLS/amf.key",
-            source=private_key.decode(),
-        )
+        self.assertEqual((root / "support/TLS/amf.key").read_text(), private_key.decode())
```

Comment:
This assert is not exactly what we are testing for.

We should test that config file was not pushed again to the container.